res/controllers/common-hid-packet-parser.js

```diff
@@ -1,1549 +1,2304 @@
+        bit.mapped_name = undefined;
+        bit.bitmask = bitmask;
+        bit.bit_offset = this.getOffset(bitmask);
+        bit.callback = undefined;
+        bit.value = undefined;
+        bit.auto_repeat = undefined;
+        bit.auto_repeat_interval = undefined;
+        this.bits[bit.id] = bit;
+    }
+    /**
+     * Add an output control bitmask to the HIDBitVector
+     *
+     * @param {string} group Control group name e.g. "[Channel1]"
+     * @param {string} name Control name e.g. "play"
+     * @param {number} bitmask A bitwise mask of up to 32 bit. All bits set to'1' in this mask are
+     *     considered.
+     */
+    addOutputMask(group, name, bitmask) {
+        /** @type {bitObject} */
+        const bit = {};
+        bit.type = "output";
+        bit.packet = undefined;
+        bit.id = `${group}.${name}`;
+        bit.group = group;
+        bit.name = name;
+        bit.mapped_group = undefined;
+        bit.mapped_name = undefined;
+        bit.bitmask = bitmask;
+        bit.bit_offset = this.getOffset(bitmask);
+        bit.callback = undefined;
+        bit.value = undefined;
+        bit.toggle = undefined;
+        this.bits[bit.id] = bit;
+    }
+}
+// Add class HIDBitVector to the Global JavaScript object
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+this.HIDBitVector = HIDBitVector;
 
-/** Add a new modifier to controller. */
-HIDModifierList.prototype.add = function(name) {
-    if (name in this.modifiers) {
-        HIDDebug("Modifier already defined: " + name);
-        return;
-    }
-    this.modifiers[name] = undefined;
-};
 
-/** Set modifier value */
-HIDModifierList.prototype.set = function(name, value) {
-    if (!(name in this.modifiers)) {
-        HIDDebug("Unknown modifier: " + name);
-        return;
-    }
-    this.modifiers[name] = value;
-    if (name in this.callbacks) {
-        var callback = this.callbacks[name];
-        callback(value);
+/**
+ * HID Modifiers object
+ *
+ * e.g. a shift button can be defined as modifier for the behavior of other controls.
+ *
+ * Wraps all defined modifiers to one object with uniform API.
+ * Don't call directly, this is available as HIDController.modifiers
+ */
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+class HIDModifierList {
+    constructor() {
+        /**
+         * Actual value of the modifier
+         *
+         * @type {Object.<string, boolean>}
+         */
+        this.modifiers = Object();
+
+        /**
+         * Function to be called after modifier value changes
+         *
+         * @type {Object.<string, modifierCallback>}
+         */
+        this.callbacks = Object();
+    }
+    /**
+     * Add a new modifier to controller.
+     *
+     * @param {string} name Name of modifier
+     */
+    add(name) {
+        if (name in this.modifiers) {
+            console.warn(`HIDModifierList.add - Modifier already defined: ${name}`);
```

Comment:
IIUC you can ask for a stacktrace, thats probably more useful than just printing which function was called. Wdyt?
```suggestion
            console.warn(`Modifier already defined: ${name}`);
            console.trace();
```

Reply:
I tried console.trace, console.assert and console.exception. All have problems (see #11080).
I suggest to change this later, when these functions work as needed.